integrations/helm/release/utils_test.go

```diff
@@ -0,0 +1,168 @@
+package release
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	ifv1 "github.com/weaveworks/flux/apis/helm.integrations.flux.weave.works/v1alpha"
+)
+
+var noopLogger = log.NewNopLogger()
+
+type mappifyTestCase struct {
```

Comment:
would it be easier to read tests if these structs are defined right before the test, or in the test case themselves (they don't need to be global)?

Reply:
It's quite common to have all the definitions at the top of the test file. I shall move them just before the tests where they are used. With more complex test setup I think it's better to have the structs defined separately and refer to them in the test cases. I consider it more readable but perhaps it's a personal thing.

Reply:
yep, if the structs are being used across multiple test cases then makes sense to define them at the top and use across multiple test cases.

otherwise it seems common to include the struct and test cases inside the test function itself. some examples:
- https://github.com/weaveworks/flux/blob/master/release/releaser_test.go#L137
- https://github.com/weaveworks/flux/blob/master/sync/sync_test.go#L69
- https://github.com/weaveworks/service/blob/master/users/tokens/tokens_test.go#L13
- https://github.com/weaveworks/cortex/blob/master/pkg/chunk/chunk_store_test.go#L105

Reply:
Is it ok to leave the tests (with the struct definition moved towards the relevant tests) this time as is?

Reply:
sure

Reply:
Thank you.